src/awkward/operations/ak_concatenate.py

```diff
@@ -18,9 +18,6 @@ def concatenate(
             outermost dimension is `0`, followed by `1`, etc., and negative
             values count backward from the innermost: `-1` is the innermost
             dimension, `-2` is the next level up, etc.
-        merge (bool): If True, combine data into the same buffers wherever
-            possible, eliminating unnecessary #ak.contents.UnionArray8_64 types
-            at the expense of materializing #ak.contents.VirtualArray nodes.
         mergebool (bool): If True, boolean and numeric data can be combined
             into the same buffer, losing information about False vs `0` and
             True vs `1`; otherwise, they are kept in separate buffers with
```

Comment:
The `merge` argument has been removed from `ak.concatenate` because now, one cannot produce an unmerged union. `mergebool` is still a meaningful argument.